frontend/src/tests/ForestClientInput.spec.ts

```diff
@@ -102,4 +121,286 @@ describe('ForestClientInput', () => {
+            'setVerifiedForestClients'
+        );
+
+        expect(wrapper.emitted()).toHaveProperty('setVerifiedForestClients');
+        expect(setVerifiedForestClients).toHaveLength(2);
+        expect(
+            setVerifiedForestClients
+                ? setVerifiedForestClients[0][0]
+                : undefined
+        ).toEqual(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        expect(
+            setVerifiedForestClients
+                ? setVerifiedForestClients[1][0]
+                : undefined
+        ).toEqual(TEST_SUCCESS_FOREST_CLIENT_NUMBER_2);
+        // the forest client information card shows and verify the information
+        expect(wrapper.html().includes('Verified Client ID information')).toBe(
+            true
+        );
+        expect(
+            wrapper
+                .find('.client-id-wrapper')
+                .text()
+                .includes(TEST_SUCCESS_FOREST_CLIENT_NUMBER)
+        ).toBe(true);
+        expect(
+            wrapper
+                .find('.org-name-wrapper')
+                .text()
+                .includes(TEST_FOREST_CLIENT_NAME)
+        ).toBe(true);
+        expect(
+            wrapper.find('.org-status-wrapper').text().includes('Active')
+        ).toBe(true);
+        // the forest client input should now be empty
+        expect(inputField.value).toBe('');
+
+        const deleteButton = wrapper.find('.custom-carbon-icon--trash-can');
+        deleteButton.trigger('click');
+        await flushPromises();
+        expect(
+            wrapper
+                .find('.client-id-wrapper')
+                .text()
+                .includes(TEST_SUCCESS_FOREST_CLIENT_NUMBER)
+        ).toBe(false);
+    });
+
+    it('should cleanup error when property change', async () => {
+        const input = wrapper.find('#forestClientInput');
+        const inputField: HTMLInputElement = input.element as HTMLInputElement;
+        await input.setValue(TEST_INACTIVE_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toBe(TEST_INACTIVE_FOREST_CLIENT_NUMBER);
+
+        await wrapper.find("[aria-label='Add Client Numbers']").trigger('click');
+        await flushPromises();
+
+        const validationError = wrapper.get('#forestClientInputValidationError');
+        const validationErrorElement: HTMLInputElement = validationError.element as HTMLInputElement;
+        // Check if the error message has been raised
+        expect(wrapper.find('#forestClientInputValidationError').element.textContent).toBeTruthy();
+
+        await input.trigger('click');
+        await flushPromises();
+
+        await input.setValue(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        // The error message should be cleared
+        expect(wrapper.find('#forestClientInputValidationError').exists()).toBeFalsy();
+    });
+    it('should cleanup forest client card when property change', async () => {
```

Comment:
This test doesn't look correct. 

So you should
1. add a success forest client number, and verify it, the forest client card displays
2. change the userId property, verify the forest client card is removed, and resetVerifiedForestClients emit method should be called 
3. repeat step 1, and change the roleId property, verify the forest client card is removed, and resetVerifiedForestClients emit method should be called

Reply:
And we need to add another test case "should cleanup forest client number when property change". 

1. add an invalid or not found forest client number, and verify it, the error displays, the number remains in the input
2. change the userId property, verify the error message removed, the input is empty, and resetVerifiedForestClients emit method should be called
3. repeat step 1, and change the roleId property, verify the error message removed, the input is empty, and resetVerifiedForestClients emit method should be called